From akpm@linux-foundation.org Mon Jul  9 12:03:59 2007
From: Venki Pallipadi <venkatesh.pallipadi@intel.com>
Date: Mon, 09 Jul 2007 12:03:06 -0700
Subject: USB: Make usb-autosuspend timer 1 sec jiffy aligned
To: greg@kroah.com
Cc: linux-usb-devel@lists.sourceforge.net, akpm@linux-foundation.org, venkatesh.pallipadi@intel.com
Message-ID: <200707091903.l69J37Jl002781@imap1.linux-foundation.org>


From: Venki Pallipadi <venkatesh.pallipadi@intel.com>

Make usb autosuspend timers 1sec jiffy aligned.

This helps to reduce the frequency at which the CPU must be taken out of a
lower-power state.

Signed-off-by: Venkatesh Pallipadi <venkatesh.pallipadi@intel.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/core/driver.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/core/driver.c
+++ b/drivers/usb/core/driver.c
@@ -1010,7 +1010,7 @@ static int autosuspend_check(struct usb_
 			 * or for the past.
 			 */
 			queue_delayed_work(ksuspend_usb_wq, &udev->autosuspend,
-					suspend_time - jiffies);
+				round_jiffies_relative(suspend_time - jiffies));
 			}
 		return -EAGAIN;
 	}
